platform-operator/controllers/verrazzano/component/mysql/mysql.go

```
@@ -60,17 +56,17 @@ func isEnabled(context spi.ComponentContext) bool {
 func appendMySQLOverrides(compContext spi.ComponentContext, _ string, _ string, _ string, kvs []bom.KeyValue) ([]bom.KeyValue, error) {
 	cr := compContext.EffectiveCR()
 
-	secret := &v1.Secret{}
-	nsName := types.NamespacedName{
-		Namespace: vzconst.KeycloakNamespace,
-		Name:      secretName}
-	// Get the mysql secret
-	err := compContext.Client().Get(context.TODO(), nsName, secret)
-	if err != nil && !k8serror.IsNotFound(err) {
-		return []bom.KeyValue{}, ctrlerrors.RetryableError{Source: ComponentName, Cause: err}
-	}
-	// see if the mysql exists
-	if !k8serror.IsNotFound(err) {
+	if compContext.For(ComponentName).GetOperation() == "upgrade" {
+		secret := &v1.Secret{}
+		nsName := types.NamespacedName{
+			Namespace: vzconst.KeycloakNamespace,
+			Name:      secretName}
+		// Get the mysql secret
+		err := compContext.Client().Get(context.TODO(), nsName, secret)
+		if err != nil {
+			compContext.Log().Errorf("Error getting mysql secret: %v", err)
+			return []bom.KeyValue{}, ctrlerrors.RetryableError{Source: ComponentName, Cause: err}
```

This is the right thing to do, but for now will still cause us to fail upgrade until we fix that.

---

Well what the desired behavior. We could either fail upgrade or log that the mysql secret was not found so the password will be reset. What do you think

---

My comment was I htink this is the right thing to do, lol.  I probably should have stayed silent :-)